### DISCUSSION TITLE
Support for chart info regarding rating scale and chart focus

### DISCUSSION THREAD

**Discussion by Sudospective** [TeamRizu/OutFox#441] [03 - Feature Requests]
Most game modes have normalized rating scales, but for some, especially dance, this can mean several types, including DDR, DDR X, ITG, and a special scale for modfiles. Now that OutFox supports many modes beyond dance, modfiles are likely to start showing up for several of these modes, which will introduce this modfile scale to those communities. I'd like to request a feature to embed the specification of a scale type in the chart data. It would also be handy to have a specification for what the focus of a chart is (Stamina, Technical, Normal, etc.) to even further demystify a chart's intent and purpose.

**Comment by Sudospective** [comment]
For example:
```
-- Example 1 --
#RATINGSCALE:DDRX;
#CHARTSTYLE:Technical;
-- Example 2 --
#RATINGSCALE:POMU;
#CHARTSTYLE:Normal;
```

**Comment by lomdr** [comment]
There used to be a tag that existed that would give a song the scale it used, but it was heavily dance centric and I forgot what it was called (something like #RATINGSCALE).  Modes that would benefit from it the most would be (ontop of Dance) BMS/newBMS (3 different Scales), Pop'n/newPop'n (2 different scales), and GDGF/GDDM.  Granted, I don't know how intense the rating scale problems are/were with those simulators.  I do know that it is a common problem in BMS for those that mix in 5key simulation with 7key simulation (bm1 used a 9 point scale while IIDX used a 7 point and then a 12 point scale)

As for chart styles, that would be better as a 'per chart' basis instead of a 'per file' basis (as some files have a mix of both Pad and Keyboard charts like in Tournamix).

**Reply by Sudospective** [reply]
Again, there are mods for a lot of modes now, so it'd be nice to know if it's using a normal scale or a modfile scale. And yes, per-chart is what I was thinking, for both scale and chart style. Some songs have only one chart that has mods in it.

**Reply by lomdr** [reply]
Yeah, true that.  And thanks for the clarification.

And yeah, maybe per chart would be better, though for those that only dabble with Official charts and some fanmixes maybe a way to batch rating scale by folder (for groups with uniform ratings like Official Mixes since X or Gpop's collections) or by song in folder (for groups that DON'T like Official mixes before X) would make things easier.

**Reply by Sudospective** [reply]
Batch scaling and styling would be handy for packs like Technical Showcase and Stamina RPG, yeah, but I think by pack would be a bit tricky to implement engine-side without adding new files. By song is definitely doable, and charts could override songs if they deviate.

**Reply by Sudospective** [reply]
> 
> 
> Modes that would benefit from it the most would be (ontop of Dance) BMS/newBMS (3 different Scales), Pop'n/newPop'n (2 different scales), and GDGF/GDDM.

I know that Guitar Hero and Rock Band have a very shared and very consistent rating scale only going to 6, but Clone Hero has a much larger one that goes past that all the way up to 99 I believe, and would likely want to be used by the community as the new meta.

**Reply by lomdr** [reply]
For Per Folder Batch Application (though it will be messy once multiple rating tags come up for multiple modes, so using Dance as an example) it could be handled almost like a 'normal' options screen (just with more room for the folder name than current) with the options available: 1-14, 1-20, Per song.  Though, to enter the Per song batch rater, you would have to hit Start on that line to enter it.  Per Song would require a more visual mark up.  For Beat it would be 1-7, 1-9, 1-12; Pomu would be 1-43, 1-50; both having a Per Song option.  Of course the operation would either edit the preexisting SSC or Ofox files or make new ones if need be.  So, maybe a caution that says "Hey, this will take some time, do not turn off the machine" would be needed

**Reply by Sudospective** [reply]
Ah, that's true, it could just be an option to batch edit rather than just a definition on one file for all others

**Reply by lomdr** [reply]
Yeah, pretty much.

As far as I know for Dance, there's pretty much the following types of charts:
Normal (Encompasses non technical DDR Style and ITG style), Stamina, Technical (DDR Style and ITG style), Gimmick [like the Radar charts in SN2], 4KAS [Four Keys, Autoscratch, so IIDX style charts for Dance] and Keyboard [some overlap with 4KAS, but not some keyboard charts are NOT that].  Though, I feel I might be missing some

**Reply by Sudospective** [reply]
There are a lot of joke charts, especially in packs like SMH, maybe that could be a listed chart style? I don't necessarily want to encourage those kinds of charts but if people are gonna make them anyway...

**Reply by lomdr** [reply]
Yeah, true that, Sudo, with joke charts.  At the same time, "Lethal Joke Charts" are a huge thing in CH that can cross over to GH mode with that mode's availability.  There's like 1 Pop Candy chart and 1 Dancing Stage Hypermix 2 chart that's padable in all difficulties except for Expert (Wan Wan March in PC Twinkle, Shonen A in DSHM2 feels like BMR's A's son), so I can see the need for the style tag/icon by chart)

Also, I remember the old field that allowed you to set what scale it used: #METERTYPE.  Though, that field might need some updating
